go/vt/vtgate/planbuilder/operators/derived.go

```diff
@@ -37,40 +35,18 @@ type Derived struct {
 	ColumnsOffset []int
 }
 
-var _ abstract.PhysicalOperator = (*Derived)(nil)
-
-// TableID implements the PhysicalOperator interface
-func (d *Derived) TableID() semantics.TableSet {
-	return d.Source.TableID()
-}
-
-// UnsolvedPredicates implements the PhysicalOperator interface
-func (d *Derived) UnsolvedPredicates(semTable *semantics.SemTable) []sqlparser.Expr {
-	return d.Source.UnsolvedPredicates(semTable)
-}
-
-// CheckValid implements the PhysicalOperator interface
-func (d *Derived) CheckValid() error {
-	return d.Source.CheckValid()
-}
+var _ PhysicalOperator = (*Derived)(nil)
 
 // IPhysical implements the PhysicalOperator interface
 func (d *Derived) IPhysical() {}
 
-// Cost implements the PhysicalOperator interface
-func (d *Derived) Cost() int {
-	return d.Source.Cost()
-}
-
-// Clone implements the PhysicalOperator interface
-func (d *Derived) Clone() abstract.PhysicalOperator {
+// Clone implements the Operator interface
+func (d *Derived) Clone(inputs []Operator) Operator {
+	checkSize(inputs, 1)
 	clone := *d
-	clone.Source = d.Source.Clone()
+	clone.Source = inputs[0]
```

Comment:
Shouldn't we call `Clone` on the input? Or is the caller already calling `Clone`?

Reply:
yeah, the idea is that the new `Clone` method gets the inputs given to it. that way, the operators don't have to think about what to do with inputs

Reply:
that's clever, okay!